src/controllers/sign_up.js

```diff
@@ -1,3 +1,64 @@
-exports.get = (req,res) =>{
-    res.render('sign_up',{ style:"style", dom:"sign-up", title:"sign up",a:true});
+const hashPassword = require('./hashPassword');
+const { addUser } = require('../model/queries/users');
+const { addPerson } = require('../model/queries/person');
+const { addBusiness } = require('../model/queries/business');
+
+exports.get = (req, res) => {
+  res.render('sign_up', {
+    style: 'style', dom: 'sign_up', vald: 'validation', title: 'sign up', a: true,
+  });
+};
+
+exports.post = (request, response) => {
+  const data = request.body;
+  const { email } = data;
+  const { password } = data;
+  const { type } = data;
+
+  if (type == 0 || 1 && email && password) {
+    hashPassword(password, (err, hash) => {
+      if (err) {
+        response.render('sign_up', { dom: 'sign_up', msg: 'ERROR' });
```

Comment:
Error in hashing the password should result a server error with status code 500.